fix(git): further fixes for submodule worktrees

pitkali opened the pull request:
filepath.Join() just concatenates paths. We have to check manually for
an absolute path and discard the first component to end up with the
correct path.

Also, fix the tests to demonstrate actual behaviour of git, not what
filepath.Join() was doing.

Additionally, note that on Windows git paths in gitdir use unix
separators, so search for `/worktrees/` instead of using system
separator, just like when processing all the other cases.

### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md]
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

### Description

I have updated the code to actually handle absolute paths correctly. The tests seemed quite confused in particular, and needed some adjustments.

Interestingly, when testing on Windows, I discovered that prior attempt to use path separator when searching for `worktrees` in the path from Git was misguided as even my official Windows distribution of Git used `/` instead of the usual separator on Windows. This is probably why all the other searches ran on working forlder path also use `/`.

Tests pass on my Linux box. I also tested it in the wild both on Linux and Windows.

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

pitkali committed:
fix(git): further fixes for submodule worktrees

filepath.Join() just concatenates paths. We have to check manually for
an absolute path and discard the first component to end up with the
correct path.

Also, fix the tests to demonstrate actual behaviour of git, not what
filepath.Join() was doing.

Additionally, note that on Windows git paths in gitdir use unix
separators, so search for `/worktrees/` instead of using system
separator, just like when processing all the other cases.

pitkali commented:
Right. The unit test fails on Windows, because the testcase uses only unix paths. The absolute paths would need `C:` prepended on Windows, so that they are absolute. Not quite sure how to do it in Go without "hacks."

JanDeDobbeleer commented:
@pitkali if the paths are always unix styled, I guess the check needs to be more explicit and validate for a leading `/`?

pitkali committed:
fix(git): better cross-platform path resolution and tests

The simple filepath.IsAbs() is not enough on Windows, where the relative
path may start with a separator and should then be taken relative to the
volume name.

So, introduce resolveGitPath() helper function that will do the right
thing on both Windows and other systems. As a bonus, it returns paths
converted to slashes for ease of use with the rest of the git segment:
- Git for Windows uses only slashes.
- Slashes do work as a path separator on Windows anyway.
- Some tests in git segment still use (and rightfully so) slashes.

Add tests for resolveGitPath() on both types of systems and fix
TestEnableInWorktree using a system-dependent testing constant as a root
path.

pitkali commented:
@JanDeDobbeleer I have finally sat down and fixed the cross-platform concerns, including the tests. It works for me on both Windows and Linux, and the tests pass on both as well.

JanDeDobbeleer commented:
@pitkali the error is a new linter check that fails. I'll fix that tomorrow morning, and will rebase your branch after.
